test/test_helper.rb

```
@@ -17,16 +17,11 @@ def initialize(options = {})
   end
 
   def wait(regex = %r{booted}, timeout = 30)
-    Timeout::timeout(timeout) do
```

Why do we need to get rid of this?

If you make a mistake coding it's possible to start infinite loops> i wanted some way to get out.

---

So `Timeout` doesn't work well with threads. I have also seen `Timeout` swallow exceptions and hide what is really happening. There is honestly not a good way to get around the use of Timeout. Ruby 2.3 was throwing an error saying it couldn't find `Timeout`, and I explicitly included it, and it worked. For giggles, I tried removing it and it still worked, so I thought that was a better course.

See: http://www.mikeperham.com/2015/05/08/timeout-rubys-most-dangerous-api/

---

I mean you could also link to the article I wrote about this ;)

https://schneems.com/2017/02/21/the-oldest-bug-in-ruby-why-racktimeout-might-hose-your-server/

This is part of the test helpers and it's very much needed.

---

Well, dang. Okay, I will revert it.